### PR TITLE
Add user guide for connecting applications

### PR DESCRIPTION
**Pull request by bprashanth**
Incomplete, but might get in before 3. Plan to modify next week though.
@bgrant0607

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks!
cc @thockin @erictune

**Comment by bgrant0607** [comment]
minor: doc name: connecting-applications.md rather than connection-applications.md.

File: docs/user-guide/connection-applications.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  type: NodePort
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+```
+
+If you have [firewall rules](../services-firewalls.md),  you will have to open the tcp port in the output of the kubectl command before you can access your service:
```
Comment (bgrant0607) [inline docs/user-guide/connection-applications.md]: This link is busted. Please check links using the new link checker.

Should be ../../docs/services-firewalls.md

**Comment by bprashanth** [comment]
Fyi my original doc that I ended up cutting quite a bit out of 
https://docs.google.com/document/d/1FuVvKPGUPI31-Xq8oiwFyJ6B-HgJxJ3C_9HU0TRgsU4/edit#

**Comment by bgrant0607** [comment]
You need to rebase

**Comment by bprashanth** [comment]
Really? just did and git didn't complain. Pushed rebased commits anyway.

**Comment by bgrant0607** [comment]
LGTM. Will look at actual content later.
ok to merge

**Comment by bgrant0607** [comment]
Did you run the link munger on this?

**Comment by bprashanth** [comment]
ran pre-commit locally and it passed

File: docs/user-guide/connecting-applications.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+You can read more about [how we achieve this](../networking.md#how-to-achieve-this) if you’re curious.
+
+## Creating a Service for the pods
+
+So we have pods running nginx in a flat, cluster wide, address space. In theory, you could talk to these pods directly, but what happens when a node dies? The pods die with it, and the replication controller will create new ones, with different ips. This is the problem a Service solves.
+
+A Kubernetes Service is an abstraction which defines a logical set of Pods running somewhere in your cluster, that all provide the same functionality. When created, each Service is assigned a unique IP address (also called clusterIP). This address is tied to the lifespan of the Service, and will not change while the Service is alive. Pods can be configured to talk to the Service, and know that communication to the Service will be automatically load-balanced out to some pod that is a member of the Service ([why not use round robin dns?](../services.md#why-not-use-round-robin-dns)).
```
Comment (thockin) [inline docs/user-guide/connecting-applications.md]: No mention of headless services?

File: docs/user-guide/connecting-applications.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+metadata:
+  name: nginxsvc
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    app: nginx
+```
+This specification will create a Service which targets TCP port 80 on any Pod with the "app=nginx" label, and expose it on a targetPort (set to the `port` by default) that other pods can use to access the Service. Check your Service:
```
Comment (thockin) [inline docs/user-guide/connecting-applications.md]: This is wrong.  port is the abstract port on the Service.  targetPort is the port on the pods

File: docs/user-guide/connecting-applications.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  type: NodePort
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+```
+
+If you have [firewall rules](../services-firewalls.md),  you will have to open the tcp port in the output of the kubectl command before you can access your Service:
```
Comment (thockin) [inline docs/user-guide/connecting-applications.md]: This is automatic now!

File: docs/user-guide/connecting-applications.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+```
+You should see a similar message informing you about firewall rules on port 80:
+```shell
+$ kubectl delete svc nginxsvc
+$ kubectl create -f nginxsvc.yaml
+An external load-balanced service was created.  On many platforms (e.g. Google Compute Engine),
```
Comment (thockin) [inline docs/user-guide/connecting-applications.md]: You won't get these warning messages any more

File: docs/user-guide/connecting-applications.md (hunk 2 above)
Comment (bgrant0607) [inline docs/user-guide/connecting-applications.md]: I planned for that to go into a doc about "advanced use cases", along with setting up your own load balancer, importing services (nil selector), session affinity, and some non-service-related topics.

File: docs/user-guide/connecting-applications.md (hunk 4 above)
Comment (bprashanth) [inline docs/user-guide/connecting-applications.md]: It works out of the box, but it doesn't look like we removed the warnings https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/create.go#L126

Comment (bgrant0607) [inline docs/user-guide/connecting-applications.md]: It also looks like we didn't update the documentation:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/services-firewalls.md

Comment (thockin) [inline docs/user-guide/connecting-applications.md]: Dagnabbit

**Pull request merged**